src/platform_impl/macos/view.rs

```diff
@@ -100,6 +98,12 @@ pub unsafe fn set_ime_position(ns_view: id, input_context: id, x: f64, y: f64) {
     let _: () = msg_send![input_context, invalidateCharacterCoordinates];
 }
 
+pub unsafe fn set_option_as_alt(ns_view: id, option_as_alt: bool) {
+    let state_ptr: *mut c_void = *(*ns_view).get_mut_ivar("winitState");
+    let state = &mut *(state_ptr as *mut ViewState);
+    state.ignore_alt_modifier = option_as_alt;
```

Comment:
This doesn't make sense to me. Why are alt modifiers ignored when option is treated as alt? That seems opposite to what would be logical.

Reply:
I think there is a terminology clash here. According to Apple documentation the event has the property charactersIgnoringModifiers, which is the raw character being pressed. See more here: https://developer.apple.com/documentation/appkit/nsevent/1524605-charactersignoringmodifiers

However, the ev_mods are associated with the window event. And the upper layers will interpret the Option key press as an Alt modifier, and thus, if `alt_send_esc` is true, will send the escape character with the raw character. So, the flag is named `ignore_alt_modifier` as an indicator that the raw character needs to be retrieved. We can change this name to `ignore_option_modifier` or `ignore_diacritical_marks` if you think that any of those are clearer?

Reply:
I think it's probably better to have a more descriptive name and state exactly what the modifier should do. The option key is alt in both cases, plenty of platforms support alternative keyboard input with alt (it's alternate after all) and raw character modification is obviously necessary to. So instead of just saying that "option is alt" which is not clear at all, we should state what the option key actually does and do the same for the other variable here.

Reply:
To add to @chrisduerr's feedback, I'd like to suggest a specific alternative name: `alt_ignored_in_text` this naming could be used throughout, e.g. the window builder would have `with_alt_ignored_in_text`.

Reply:
@ArturKovacs What exactly is "alt ignored in text" supposed to mean?

Reply:
It's supposed to mean that text coming from the keyboard (ie `ReceivedCharacters`) will not be affected by <kbd>alt</kbd>. In other words, text ignores the <kbd>alt</kbd> key.

Reply:
I think assuming that people will make the connection between "text" and the ReceivedCharacter event is a bit much. I certainly wasn't able to.

Reply:
Consider accompanying your critiques with suggested solutions. In my opinion that speeds up such discussions.

And what do you think about `alt_ignored_in_chars`?

Reply:
You can just call it `alt_dead_keys` or whatever. But style questions like these are really irrelevant as long as the functionality doesn't even work yet.